### DISCUSSION TITLE
equivalent of express `.res.locals`

### DISCUSSION THREAD

**Discussion by ahmafi** [honojs/hono#1011] [Q&A] [answered]
Are [set/get](https://hono.dev/api/context#set-get) methods equivalent of express [`.res.locals`](https://stackoverflow.com/questions/18875292/passing-variables-to-the-next-middleware-using-next-in-express-js)?

I want to share something from a middleware to another middleware or `next()` route function.

**Comment by yusukebe** [comment, marked as answer]
Hi @ahmafi 

The answer is probably yes. Try this:

```ts
type Variables = {
  message: string
}

const app = new Hono<{ Variables: Variables }>()

app.use('*', async (c, next) => {
  c.set('message', 'This is a message')
  await next()
})

app.get('/', (c) => {
  const message = c.get('message')
  return c.text(message) // Hello!
})
```

**Reply by ahmafi** [reply]
Great. So these are set per request and then they are deleted, right?

**Reply by yusukebe** [reply]
Hi @ahmafi !

Yes, per request.
